src/main/java/com/amihaiemil/camel/NoCommentsYamlLine.java

```diff
@@ -56,22 +56,21 @@ public int compareTo(final YamlLine other) {
 
     @Override
     public String trimmed() {
-        String string = this.line.trimmed();
+        String trimmed = this.line.trimmed();
         int i = 0;
-        while(i < string.length()) {
-            if(string.charAt(i) == '#') {
-                string = string.substring(0, i);
+        while(i < trimmed.length()) {
+            if(trimmed.charAt(i) == '#') {
+                trimmed = trimmed.substring(0, i);
                 break;
-            } else if(string.charAt(i) == '"') {
+            } else if(trimmed.charAt(i) == '"') {
                 i++;
-                while(i < string.length() && string.charAt(i) != '"') {
+                while(i < trimmed.length() && trimmed.charAt(i) != '"') {
                     i++;
                 }
-            } else {
-                i++;
             }
+            i++;
```

Comment:
@amihaiemil I'm really confused but that is what I'm thinking at, there are 2 cases: 
First: it's a `"` char and the while loop will stop when it finds another one but `i` will not be incremented so the modified test fails. 
Second: it's any other char so i should be incremented. 
What's your opinion :D ?